packages/camera/camera/example/ios/RunnerTests/CameraMethodChannelTests.m

```diff
@@ -0,0 +1,101 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+@import camera;
+@import XCTest;
+@import AVFoundation;
+#import <OCMock/OCMock.h>
+
+@interface FLTThreadSafeFlutterResult ()
+@property(readonly, nonatomic) FlutterResult flutterResult;
+@end
+
+@interface MockFLTThreadSafeFlutterResult : FLTThreadSafeFlutterResult
+@property(readonly, nonatomic) NSNotificationCenter *notificationCenter;
+@property(nonatomic, nullable) id receivedResult;
+@end
+
+@implementation MockFLTThreadSafeFlutterResult
+/**
+ Initialize with a notification center.
+ */
+- (id)initWithNotificationCenter:(NSNotificationCenter *)notificationCenter {
+  self = [super init];
+  _notificationCenter = notificationCenter;
+  return self;
+}
+
+/**
+ Called when result is successful. Sends "successWithData" to the notification center.
+ */
+- (void)successWithData:(id)data {
```

Comment:
Shouldn't there be a thread assertion in this code?

Reply:
What do you mean with the thread assertion? That we verify that it is not the main thread?

Reply:
I don't remember what I meant by having that comment in this location unfortunately; maybe I was forgetting the flow and thinking this was after the bounce back to the main thread.

However, the more general questions is still valid: the goal of this PR is to ensure that methods are getting callbacks on the main thread, so why isn't there any code in the test that calls into the camera handler with a result object controlled by the test, and actually asserts that the callback was on the main thread? Like this: https://github.com/flutter/plugins/blob/master/packages/local_auth/example/ios/RunnerTests/FLTLocalAuthPluginTests.m#L62-L64

It is not clear to me what in this test would deterministically fail without the fix from this PR.

Reply:
@stuartmorgan I have now added ThreadSafeFlutterResultTests that test that the result is always called on the main thread. These tests in CameraMethodChannelTests just verify that ThreadSafeFlutterResult is used, not how that is working.